16307 - Restore ability for reports to be able to log_success without generating a message

Julio-Oliveira-Encora opened the pull request:
### Fixes: #16307

It was added to bypass the log message when the status is success, and the message is None.

It was tested using `log_success(None, None)`,  `log_success("", None)`, and `log_sucess()`.

Julio-Oliveira-Encora committed:
Added check in views.py to bypass the log messages if the status is success and message is None.

Fixed issue related to edit script ("AttributeError: 'dict' object has no attribute 'append'")

arthanson commented on netbox/extras/signals.py:
```diff
@@ -118,9 +118,7 @@ def handle_changed_object(sender, instance, **kwargs):
-    """
-    Fires when an object is deleted.
-    """
+    """Fires when an object is deleted."""
```
Can you remove this from the PR?  Nothing to do with the chagne.

arthanson commented on netbox/extras/views.py:
```diff
@@ -1195,6 +1195,7 @@ def get_table(self, job, request, bulk_actions=True):
                     tests = job.data['tests']
 
                 for log in job.data['log']:
+                    print(log)
```
Remove this print statement

Julio-Oliveira-Encora committed:
Addressed PR comments

arthanson commented:
@Julio-Oliveira-Encora it looks like log_success() works now with this change, can you please add a note to the documentation on scripts about this.

Julio-Oliveira-Encora committed:
Merge branch 'develop' into 16307-Restore_ability_for_reports_to_be_able_to_log_success_without_generating_a_message

Julio-Oliveira-Encora committed:
Addressed PR comment.

jeremystretch reviewed (changes requested):
@Julio-Oliveira-Encora have you tested this before submitting the PR? You haven't actually updated [the method's signature](https://github.com/netbox-community/netbox/blob/eb3d4230778f26fa5be38c7d6b3e1205f905fc93/netbox/extras/scripts.py#L513) to make `message` an optional argument. When calling `log_success()` in a script, an exception is raised:

```
Traceback (most recent call last):
  File "/home/jstretch/projects/netbox/netbox/extras/scripts.py", line 658, in _run_script
    script.output = script.run(data, commit)
  File "/home/jstretch/projects/netbox/netbox/scripts/test2.py", line 19, in run
    self.log_success()
TypeError: BaseScript.log_success() missing 1 required positional argument: 'message'
```

Julio-Oliveira-Encora commented:
I tested it before submitting the PR. I will check the method's signature and test again.

Julio-Oliveira-Encora committed:
Merge branch 'develop' into 16307-Restore_ability_for_reports_to_be_able_to_log_success_without_generating_a_message

Julio-Oliveira-Encora committed:
Addressed PR comment.

jeremystretch reviewed (changes requested):
I'm unclear on what the intended goal is here. As it stands, calling `log_success(obj=thing)` in a custom script does not generate any output. This seems completely pointless, and confusing.

#16307 seems limited to incrementing success counters in _reports_ only; if that's the case, the `log_*()` methods on the Report class should be modified to call `self._log()` directly rather than proxying through the `log_*()` methods on Script, to avoid changing them.

Julio-Oliveira-Encora commented:
@jeremystretch About the log methods on the Report class, should I change them this way?
   ```
 def log_success(self, obj=None, message=None):
        self._log(message, obj, level=LogLevelChoices.LOG_SUCCESS)
```
Or follow the argument positional from scripts, where `message` is before `obj`?
   ```
 def log_success(self, message, obj=None):
        self._log(message, obj, level=LogLevelChoices.LOG_SUCCESS)
```

Julio-Oliveira-Encora commented:
We could discuss it more because the user is asking for the change for Reports, but maybe he wants to use it in all types of scripts.

Julio-Oliveira-Encora committed:
Addressed PR comment.

The logs for Report class logs were changed to keep the compatibility with the documentation.

jeremystretch commented:
@Julio-Oliveira-Encora I think we went down the wrong path with this a bit. I've submitted an alternate PR (#16697) which I believe addresses the proposal in #16307.
